### PR TITLE
Stable order of parsed plugin declarations

### PR DESCRIPTION
**Pull request by v-chernyshev**
Fixes #138

Let's sort/deduplicate the plugins by alias. This way, the order of automatically added plugin marker dependencies will no longer depend on the inherently unstable FileTree iteration behaviour.

### CODE REVIEW

File: src/main/kotlin/dev/panuszewski/gradle/versioncatalogs/PluginVersionCatalogAccessors.kt [hunk 1]
```diff
@@ -40,7 +40,7 @@ private fun collectPluginDeclarations(project: Project, catalog: VersionCatalogB
         .map(::removeComments)
         .flatMap(String::lines)
         .mapNotNull { line -> parsePluginDeclaration(line, model) }
-        .toList()
+        .toSortedSet(compareBy { it.pluginAlias })
```
Comment (radoslaw-panuszewski) [inline src/main/kotlin/dev/panuszewski/gradle/versioncatalogs/PluginVersionCatalogAccessors.kt]:
```suggestion
        .sortedBy { it.pluginAlias }
```
I would prefer to keep it a `List` instead of converting to `Set` since the semantics of `List` interface imply defined order. 

According to JavaDoc from `java.util.List`:
> An ordered collection (also known as a sequence).

With this change, let's also change back from `Collection<PluginDeclaration>` to `List<PluginDeclaration>` in other places in this file 😉

**Review by radoslaw-panuszewski — COMMENTED** [review]
Hey @v-chernyshev and thanks a lot for this contribution! ❤️ 

I left two small comments, and I'll be happy to release a new version as soon as this PR is merged 😉

File: src/main/kotlin/dev/panuszewski/gradle/versioncatalogs/PluginVersionCatalogAccessors.kt (hunk 1 above)
Comment (radoslaw-panuszewski) [reply src/main/kotlin/dev/panuszewski/gradle/versioncatalogs/PluginVersionCatalogAccessors.kt]: Would you mind adding a test for the case with duplicates?

**Review by radoslaw-panuszewski — APPROVED** [review]
